audio/softsynth/mt32.cpp

```diff
@@ -169,12 +169,13 @@ int MidiDriver_MT32::open() {
 	debug(4, _s("Initializing MT-32 Emulator"));
 
 	Common::File controlFile;
-	if (!controlFile.open("CM32L_CONTROL.ROM") && !controlFile.open("MT32_CONTROL.ROM"))
-		error("Error opening MT32_CONTROL.ROM / CM32L_CONTROL.ROM. Check that your Extra Path in Paths settings is set to the correct directory");
-
 	Common::File pcmFile;
-	if (!pcmFile.open("CM32L_PCM.ROM") && !pcmFile.open("MT32_PCM.ROM"))
-		error("Error opening MT32_PCM.ROM / CM32L_PCM.ROM. Check that your Extra Path in Paths settings is set to the correct directory");
+	if (!controlFile.open("CM32L_CONTROL.ROM") || !pcmFile.open("CM32L_PCM.ROM")) {
```

Comment:
```suggestion
	if (!controlFile.open("CM32L_CONTROL.ROM") && !pcmFile.open("CM32L_PCM.ROM")) {
```

Reply:
I suspect that the original code is OK.
Using `&&` will make it fail only if both files are not found.
If only one is found it also should fail.

Reply:
> I suspect that the original code is OK. Using `&&` will make it fail only if both files are not found. If only one is found it also should fail.

Well, if any of the two files can't be opened, the check should fail. The following is the equivalent way of writing the same (edited) check:

```suggestion
	if (!(controlFile.open("CM32L_CONTROL.ROM") || pcmFile.open("CM32L_PCM.ROM"))) {
```

Reply:
Sorry, but I think this is not correct.
What we want is both files open. If it's not the case we must fall back.
This equates to:
```suggestion
	if (!(controlFile.open("CM32L_CONTROL.ROM") && pcmFile.open("CM32L_PCM.ROM"))) {
```
If you apply De Morgan's law, you get this:
```suggestion
	if (!controlFile.open("CM32L_CONTROL.ROM") || !pcmFile.open("CM32L_PCM.ROM")) {
```

Reply:
You're correct. The original code is OK here